File: Recursion/Factorial.py
#recursive
def Factorial(val: int) -> int:
    if val <= 1:
        return 1
    
    return val * Factorial(val - 1)

print(Factorial(120))

#iteration
def Factorial_It(val: int) -> int:

    res = 1
    while val > 1:
        res = val * res
        val = val - 1
    
    return res

print(Factorial_It(5))

File: Playground/Factorial.c
#include <stdio.h>
#include <stdlib.h>

int factorialRecursive(int num)
{
    if(num == 0) // base case
        return 1;
    else
        return num * factorialRecursive(num-1);
}
int factorialBrute(int num)
{
    int res=1;
    if(num < 0)
        return 0;

    while(num > 0)
    {
        res = res * num;
        num--;
    }
    return res;
}

int main()
{
    int num = 7;

    printf("Recursive: %d\r\n", factorialRecursive(num));
    printf("Brute Force: %d\r\n", factorialBrute(num));

    return 0;
}